src/api/spec/features/webui/search_spec.rb

```diff
@@ -0,0 +1,332 @@
+    expect(page).to have_text("Advanced")
+    expect(page).to have_current_path(search_path, only_path: true)
+  end
+
+  scenario "basic search functionality" do
+    package
+    reindex_for_search
+
+    visit search_path
+    expect(page).to have_current_path(search_path, only_path: true)
+
+    fill_in 'search_input', with: package.name
+    click_button 'search_button'
+
+    expect(page).to have_current_path(search_path, only_path: true)
+    within '#search_results' do
+      expect(page).to have_link(user.home_project_name)
+      expect(page).to have_link(package.name)
+    end
+  end
+
+  scenario "header search functionality" do
+    package
+    reindex_for_search
+
+    visit root_path
+
+    fill_in 'search', with: package.name
+    page.evaluate_script("$('#global-search-form').get(0).submit()")
+
+    expect(page).to have_current_path(search_path, only_path: true)
+    within '#search_results' do
+      expect(page).to have_link(user.home_project_name)
+      expect(page).to have_link(package.name)
+    end
+  end
+
+  scenario "search by request number" do
+    bs_request
+    reindex_for_search
+
+    visit root_path
+
+    fill_in 'search', with: "##{bs_request.id}"
+    page.evaluate_script("$('#global-search-form').get(0).submit()")
+
+    expect(page).to have_current_path(request_show_path(bs_request))
+    expect(page).to have_text("Request #{bs_request.id}")
+  end
+
+  describe "search by baseurl" do
+    before do
+      package
+      reindex_for_search
+
+      visit root_path
+    end
+
+    scenario "with an existing package url" do
+      fill_in 'search', with: "obs://#{user.home_project_name}/#{user.home_project.name}/i586/1-#{package.name}"
+      page.evaluate_script("$('#global-search-form').get(0).submit()")
+
+      expect(page).to have_current_path(package_show_path(user.home_project, package, rev: 1))
+      expect(page).to have_text("Source Files")
+    end
+
+    scenario "with a non existing package url" do
+      fill_in 'search', with: "obs://fake_url"
+      page.evaluate_script("$('#global-search-form').get(0).submit()")
+
+      expect(page).to have_current_path(root_path, only_path: true)
+      expect(find('#flash-messages')).to have_text('Sorry, this disturl does not compute...')
+    end
+  end
+
+  scenario "search for home projects" do
+    apache
+    tom
+    henne
+    reindex_for_search
+
+    visit search_path
+    expect(page).to have_current_path(search_path, only_path: true)
+
+    fill_in 'search_input', with: 'Home'
+    click_button 'Advanced'
+    check 'project'
+    uncheck 'package'
+    click_button 'search_button'
+
+    expect(page).to have_current_path(search_path, only_path: true)
+    within '#search_results' do
+      expect(page).to have_link(tom.home_project_name)
+      expect(page).to have_link(henne.home_project_name)
+      expect(page).to have_selector('.search_result', count: 2)
+    end
+  end
+
+  scenario "search for projects and subprojects" do
```

Comment:
Looks like... I will get rid of the above test... this is "better" in the way of searching substrings in the  middle, not just in the beginning of the string